AutoFlow_OmicsDataHandling/thermo/io.py

```diff
@@ -0,0 +1,134 @@
+"""I/O.
+
+This module hosts functions to load cobra models, create and adjust tFBA-ready
+models.
+"""
+
+import os.path
+
+from cobra.io import load_json_model
+from pytfa import ThermoModel
+from pytfa.io import (
+    load_thermoDB,
+    read_lexicon,
+    read_compartment_data,
+    annotate_from_lexicon,
+    apply_compartment_data,
+)
+
+from AutoFlow_OmicsDataHandling.thermo import static
+
+
+def _path(filename):
+    # Returns the path to a file in the static folder
+    return os.path.join(os.path.dirname(static.__file__), filename)
+
+
+def load_cbm(model_name):
+    """
+    Load a JSON cobra model stored in the static folder.
+
+    Parameters
+    ----------
+    model_name : str
+        The name of the model.
+
+    Returns
+    -------
+    cobra.Model
+        The loaded cobra model.
+    """
+    return load_json_model(_path(model_name + ".json"))
+
+
+def load_data(model_name):
+    """
+    Loads pre-curated model-specific thermodynamic information.
+
+    Parameters
+    ----------
+    model_name : str
+        The name of a model.
+
+    Returns
+    -------
+    thermo_data : dict
+        A thermodynamic database.
+    lexicon : pandas.DataFrame
+        A dataframe linking metabolite IDs to SEED compound IDs.
+    compartment_data : dict
+        A dictionary with information about each compartment of the model.
+    """
+    thermo_data = load_thermoDB(_path("thermo_data.thermodb"))
+    lexicon = read_lexicon(_path(os.path.join(model_name, "lexicon.csv")))
+    compartment_data = read_compartment_data(
+        _path(os.path.join(model_name, "compartment_data.json"))
+    )
+    return thermo_data, lexicon, compartment_data
+
+
+def create_model(model_name, thermo_data, lexicon, compartment_data) -> ThermoModel:
+    """
+    Creates a tFBA-ready model.
+
+    Parameters
+    ----------
+    model_name : str
+        The name of a model
+    thermo_data : dict
+        A thermodynamic database
+    lexicon : pandas.DataFrame
+        A dataframe linking metabolite IDs to SEED compound IDs
+    compartment_data : dict
+        A dictionary containing information about each compartment of the model
+
+    Returns
+    -------
+    pytfa.ThermoModel : dict
+        A thermodynamic database
+    """
+    model = load_cbm(model_name)
+    tmodel = ThermoModel(thermo_data, model)
+    tmodel.name = model_name
+
+    annotate_from_lexicon(tmodel, lexicon)
+    apply_compartment_data(tmodel, compartment_data)
+
+    if tmodel.solver.interface.__name__ == "optlang.gurobi_interface":
```

Comment:
Is there any mention of the "hard" requirement to have gurobi as the solver in the documentation? I am curious how you are working around this in the unit tests as I don't think CircleCI has gurobi.

Reply:
Sadly, I have not been able to work around it. CircleCI uses the GLPK (free, open-source) solver; and, there are two major problems with this. 1) GLPK is slow. Even using a "small_ecoli" model, the unit test times out after 10 min. 2) GLPK cannot solve the relaxation problems. For this reason, the "relax_lc" function I wrote is entirely dependent on the Gurobi interface and its unit test only checks if a ModuleNotFoundException is raised by the lack of Gurobipy.

Reply:
I see.  Svetlana V mentioned a similar issue.  I forget what her workaround was.  If you can make a new Jira issue for integration of Gurobi with the CI as a reminder, that would be great.